content/en/blog/_posts/2022-11-28-advancements-in-traffic-engineering/index.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "Kubernetes v1.26: Advancements in Kubernetes Traffic Engineering"
+date: 2022-11-28
+slug: advancements-in-kubernetes-traffic-engineering
+---
+
+**Authors:** Andrew Sy Kim (Google)
+
+Kubernetes v1.26 includes significant advancements in traffic engineering with the graduation of the
+`ServiceInternalTrafficPolicy` and `EndpointSliceTerminatingCondition` features to GA and the
+`ProxyTerminatingEndpoints` feature to Beta. The combination of these enhancements aim to address
+short-comings in traffic engineering today and unlock new capabilities for the future.
+
+## Traffic Loss from Load Balancers During Rolling Updates
+
+Prior to Kubernetes v1.26, [clusters can experience loss of traffic](https://github.com/kubernetes/kubernetes/issues/85643)
+from Service load balancers during rolling updates when setting the `externalTrafficPolicy` field to `Local`.
+There are a lot of moving parts at play here so a quick overview of how Kubernetes manages load balancers might help!
+
+In Kubernetes, you can create a Service with Type=LoadBalancer to expose an application externally with a load balancer.
+The load balancer implementation varies between clusters and platforms, but the Service provides a generic abstraction
+representing the load balancer that is consistent across all Kubernetes installations.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app.kubernetes.io/name: my-app
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  type: LoadBalancer
+```
+
+Under the hood, Kubernetes allocates a NodePort for the Service, which is then used by kube-proxy to provide a
+network data path from the NodePort to the Pod. A controller running in the Kubernetes Control Plane will then
+add all available Nodes in the cluster to the load balancer’s backend pool, using the designated NodePort for
+the Service as the backend target port.
+
+{{< figure src="traffic-engineering-service-load-balancer.png" alt="Figure 1: Overview of Service Load Balancers" >}}
```

Comment:
(nit) You can caption the image rather than setting the figure name only in the `alt` attribute.